tests/plenary/async_lib/read_file_bench.lua

```diff
@@ -0,0 +1,71 @@
+local a = require('plenary.async_lib')
+local async = a.async
+local uv = vim.loop
+
+local plenary_init = vim.api.nvim_get_runtime_file('lua/plenary/init.lua', false)[1]
+local plenary_dir = vim.fn.fnamemodify(plenary_init, ":h:h:h")
+local assets_dir = plenary_dir .. '/' .. 'tests/plenary/async/assets/'
+
+local read_file = async(function(path)
+  local err, fd = a.wait(a.uv.fs_open(path, "r", 438))
+  assert(not err, err)
+
+  local err, stat = a.wait(a.uv.fs_fstat(fd))
+  assert(not err, err)
+
+  local err, data = a.wait(a.uv.fs_read(fd, stat.size, 0))
+  assert(not err, err)
+
+  local err = a.wait(a.uv.fs_close(fd))
+  assert(not err, err)
+end)
+
+local read_file_other = function(path, callback)
+  uv.fs_open(path, "r", 438, function(err, fd)
+    assert(not err, err)
+    uv.fs_fstat(fd, function(err, stat)
+      assert(not err, err)
+      uv.fs_read(fd, stat.size, 0, function(err, data)
+        assert(not err, err)
+        uv.fs_close(fd, function(err)
+          assert(not err, err)
+          return callback(data)
+        end)
+      end)
+    end)
+  end)
+end
+
+local first_bench = async(function()
+
+  local futures = {}
+
+  local start = os.clock()
+
+  for i = 1, 200 do futures[i] = read_file(assets_dir .. 'README.md') end
```

Comment:
I actually would prefer not to ship does files in `assers_dir` just to reduce bloat (a lot of people have it installed so we shouldn't include unnecessary stuff).

But if you still wanna do a benchmark also in CI we can add some stuff to make it work. Here is what i am thinking:
- Maybe add a simple_busted.bench that can abstract some things and has a different output
- Easy comparison with a range
- We can curl a big file in CI into `assets_dir` and have a `.gitkeep` in assets_dir to keep the directory. 
  - We could use nvim binary already pulled in CI job so we could just read that one
- And some other stuff. idk thoughts?